dali/python/nvidia/dali/types.py

```diff
@@ -74,12 +74,12 @@ def _not_implemented(val):
     _known_types[DALIDataType._FEATURE_DICT] = ("dict of (string, nvidia.dali.tfrecord.Feature)",
             _not_implemented)
 
-def _type_name_convert_to_string(dtype, is_tensor):
+def _type_name_convert_to_string(dtype, allow_tensors):
     if dtype in _known_types:
         ret = _known_types[dtype][0]
-        if is_tensor:
-            ret = "TensorList of " + ret
-        elif dtype in _vector_types:
+        if allow_tensors:
+            ret = "TensorList of " + ret + " or " + ret
+        if dtype in _vector_types:
```

Comment:
Is the change from elif to if in purpose? Now  you can potentially override what was set in the previous condition

Reply:
No, it will append " or list of X", won't it?

Reply:
true, I missed that